SYCL/ESIMD/api/functional/operators/operator_logical_not.cpp

```diff
@@ -0,0 +1,161 @@
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd logical not operator.
+// The test creates source simd instance with reference data and invokes logical
+// not operator.
+// The test verifies that data from simd is not corrupted after calling logical
+// not operator, that logical not operator return type is as expected and
+// logical not operator result values is correct.
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling logical not operator.
+struct logical_not_operator {
+  static std::string get_description() { return "logical not"; }
+
+  template <typename DataT, int NumElems, typename OperatorResultT>
+  static bool call_operator(const DataT *const ref_data, DataT *const out,
+                            OperatorResultT *const operator_result) {
+    auto simd_obj = simd<DataT, NumElems>();
+    simd_obj.copy_from(ref_data);
+    auto logical_not_result = !simd_obj;
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      operator_result[i] = logical_not_result[i];
+    }
+
+    simd_obj.copy_to(out);
+    return true;
+  }
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename TestCaseT, typename DataT, typename DimT> class run_test {
+  static constexpr int NumElems = DimT::value;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    std::cout << "start test with " << data_type << std::endl;
+    bool passed = true;
+    const std::vector<DataT> ref_data = generate_ref_data<DataT, NumElems>();
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+    std::cout << "in run_verification" << std::endl;
```

Comment:
Just a friendly reminder, I'm sure you are aware. Here and elsewhere

Reply:
thanks for the reminder